src/test/java/api/loans/CheckInByBarcodeTests.java

```diff
@@ -676,6 +676,102 @@ public void overdueFineShouldBeChargedWhenItemIsOverdue() {
+      .withPickupServicePointId(UUID.fromString(homeLocation.getJson().getString("primaryServicePoint")))
+      .withTags(new RequestBuilder.Tags(asList("new", "important"))));
+
+    JsonObject servicePointOwner = new JsonObject();
+    servicePointOwner.put("value", homeLocation.getJson().getString("primaryServicePoint"));
+    servicePointOwner.put("label", "label");
+    UUID ownerId = UUID.randomUUID();
+    feeFineOwnersClient.create(new FeeFineOwnerBuilder()
+      .withId(ownerId)
+      .withOwner("fee-fine-owner")
+      .withServicePointOwner(Collections.singletonList(servicePointOwner))
+    );
+
+    UUID feeFineId = UUID.randomUUID();
+    feeFinesClient.create(new FeeFineBuilder()
+      .withId(feeFineId)
+      .withFeeFineType("Overdue fine")
+      .withOwnerId(ownerId)
+    );
+
+    loansFixture.checkInByBarcode(new CheckInByBarcodeRequestBuilder()
+      .forItem(nod)
+      .on(new DateTime(2020, 3, 10, 12, 0, 0, DateTimeZone.UTC))
+      .at(checkInServicePointId));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(accountsClient::getAll, hasSize(1));
+
+    List<JsonObject> createdAccounts = accountsClient.getAll();
+
+    assertThat("Fee/fine record should be created", createdAccounts, hasSize(1));
+
+    JsonObject account = createdAccounts.get(0);
+    assertThat("owner ID is included",
+      account.getString("ownerId"), is(ownerId.toString()));
+    assertThat("fee/fine ID is included",
+      account.getString("feeFineId"), is(feeFineId.toString()));
+    assertThat("amount is correct", account.getDouble("amount"), is(10.0));
+    assertThat("remaining is the same as amount",
+      account.getDouble("remaining"), is(10.0));
+    assertThat("correct fee/fine type is included",
+      account.getString("feeFineType"), is("Overdue fine"));
+    assertThat("fee/fine owner is included",
+      account.getString("feeFineOwner"), is("fee-fine-owner"));
+    assertThat("item's title is included",
+      account.getString("title"), is(loan.getJson().getJsonObject("item").getString("title")));
+    assertThat("item's barcode is included",
+      account.getString("barcode"), is(nod.getJson().getString("barcode")));
+    assertThat("call number from the item is included",
+      account.getString("callNumber"),
+      is(nod.getJson().getJsonObject("effectiveCallNumberComponents").getString("callNumber")));
+    assertThat("effective location ID is included",
+      account.getString("location"),
+      is(servicePointsFixture.cd1().getId()));
+    assertThat("item's material type is included",
+      account.getString("materialTypeId"),
+      is(nod.getJson().getString(ItemProperties.MATERIAL_TYPE_ID)));
+    assertThat("loan ID is included", account.getString("loanId"), is(loan.getId()));
+    assertThat("user ID is included",
+      account.getString("userId"), is(loan.getJson().getString("userId")));
+    assertThat("item ID is included", account.getString("itemId"), is(nod.getId()));
```

Comment:
Could this be extracted into a method or matcher to better express what is being checked.

Is it that the expected fine is charged?

Reply:
Yes. Moved this to a matcher.